test/gtest/uct/ib/test_ib_pkey.cc

```diff
@@ -6,18 +6,20 @@
 #include <uct/ib/test_ib.h>
 
 
-class test_uct_ib_pkey : public test_uct_ib {
+class test_uct_ib_pkey : public test_uct_ib_with_specific_port {
 public:
-    void init() {
-        test_uct_ib::init();
-
+    void check_port_attr() {
         if (IBV_PORT_IS_LINK_LAYER_ETHERNET(&m_port_attr)) {
-            test_uct_ib::cleanup();
             /* no pkeys for Ethernet */
             UCS_TEST_SKIP_R("skip pkey test for port with Ethernet link type");
         }
     }
 
+    void send_recv_short() {
+        create_connected_entities();
+        test_uct_ib::send_recv_short();
```

Comment:
we need to destroy the entities after the test, what if this function is called many times, it means a lot of resources will be consumed?

Reply:
yes, this is a good point
we able to have 256 pkey configured that will create 512 entities

done